serde/src/de/impls.rs

```diff
@@ -297,6 +345,19 @@ macro_rules! uint_to_self {
     };
 }
 
+#[cfg(not(no_core_num_saturating))]
+macro_rules! visit_saturating {
+    ($ty:ident : $visit:ident) => {
+        #[inline]
+        fn $visit<E>(self, v: $ty) -> Result<Self::Value, E>
+        where
+            E: Error,
+        {
+            Ok(convert_saturating(v))
+        }
+    };
+}
```

Comment:
I am always surprised by the ordering shenanigans that macros allow. I'd prefer this to live above the `impl_deserialize_saturating_num` macro